src/raft.rs

```diff
@@ -1072,9 +1072,22 @@ impl<T: Storage> Raft<T> {
                 let to_send = new_message(m.from, MessageType::MsgAppendResponse, None);
                 self.send(to_send);
             } else if m.get_msg_type() == MessageType::MsgRequestPreVote {
-                // Before pre_vote enable, there may be a recieving candidate with higher term,
-                // but less log. After update to pre_vote, the cluster may deadlock if
-                // we drop messages with a lower term.
+                // Unlike normal election, both sender and receiver of MsgReqPreVote{Resp}
```

Comment:
I think the original comment is quite sufficient. The comment in this PR may give a wrong impression that the check is for a rarely happen situation.

Reply:
Maybe just for me but it took me a while to figure out what the deadlock is. :)
Surely the example is misleading a verbose. How about I remote that part?